pramen-py/src/pramen_py/metastore/reader.py

```diff
@@ -57,6 +57,28 @@ def _apply_uppercase_to_columns_names(
         else:
             return df
 
+    def _extract_dates_from_files_name(
+        self,
+        files: List[str],
+        date_format: str,
+    ) -> List[datetime.date]:
+        def is_file_hidden(column_name_: str, date_: str) -> bool:
+            if len(column_name_) > 0 and column_name_[0] == "_":
+                return True
+            if len(date_) > 0 and date_[0] == "_":
+                return True
+            return False
+
+        extracted_dates = []
+        for path in files:
+            file_name = path.rsplit("/", 1)[-1]
+            column_name, date = file_name.split("=")
+            if not is_file_hidden(column_name, date):
+                extracted_dates.append(
+                    convert_str_to_date(date, fmt=date_format)
+                )
```

Comment:
Consider extracting this logic as a separate function, say `_extract_date_from_file_name() -> Optional[datetime.date]`.

This way the code might look like this:
```python
def _extract_dates_from_file_names(...):
   extracted_dates = []
   for path in files:
     dateOpt = _extract_date_from_file_name(path, ...)
     if dateOps is not None:
        extracted_dates.append(date.get())
   return extracted_dates = []
```